app/controllers/songs_controller.rb

```diff
@@ -1,25 +1,44 @@
 class SongsController < ApplicationController
+
+  skip_before_action :verify_authenticity_token
+  before_action :set_song, only: [:show, :update, :destroy]
+
   def index
     @songs = Song.all
 
-    render json: @songs
+    render json: @songs, each_serializer: SongIndexSerializer
+  end
+
+  def show
+    render json: @song, each_serializer: SongShowSerializer
   end
 
   def create
-    @song = song.new(song_params)
+    @song = Song.new(song_params)
 
     @song.save
```

Comment:
Use [it](https://apidock.com/rails/ActiveResource/Base/persisted%3F) when you saving a new object in the model.